boards/arm/stm32wb5mmg/doc/stm32wb5mmg.rst

```diff
@@ -0,0 +1,304 @@
+================================================================
+
+To operate bluetooth on STM32WB5MMG, Cortex-M0 core should be flashed with
+a valid STM32WB Coprocessor binaries (either 'Full stack' or 'HCI Layer').
+These binaries are delivered in STM32WB Cube packages, under
+Projects/STM32WB_Copro_Wireless_Binaries/STM32WB5x/
+For compatibility information with the various versions of these binaries,
+please check `modules/hal/stm32/lib/stm32wb/hci/README`_
+in the hal_stm32 repo.
+Note that since STM32WB Cube package V1.13.2, "full stack" binaries are not compatible
+anymore for a use in Zephyr and only "HCI Only" versions should be used on the M0
+side.
+
+Connections and IOs
+===================
+
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+.. rst-class:: rst-columns
+
+- UART_1 TX/RX : PB7/PB6
+- LPUART_1 TX/RX : PA3/PA2
+- USB : PA11/PA12
+- SWD : PA13/PA14
+
+System Clock
+------------
+
+STM32WB5MMG System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by HSE clock at 32MHz.
+
+Serial Port
+-----------
+
+STM32WB5MMG board has 2 (LP)U(S)ARTs. LPUART1 is connected to the main U585I
+microcontroller that is used as HCI controller port. USART1 is not connected
+to any external pinout, so it is not possible to debug the module directly.
+Rather, users can use the available USB port (CN12) to run virtual com port
+(VCP) USB stack for the debugging.
+
+
+Programming and Debugging
+*************************
+
+Applications for the ``stm32wb5mmg`` board configuration can be built the
+usual way (see :ref:`build_an_application`).
+
+Flashing the module
+===================
+
+The onboard ST-Link on the ``b_u585i_iot02a`` board can be used to flash the
+STM32WB5MMG module. To do this you should put SW4 on OFF and SW5 on ON mode.
+In this case the firmware will be uploaded on the STM32WB5MMG module.
+
+
+Flashing `hci_uart` application to STM32WB5MMG
+----------------------------------------------
+
+Connect the B-U585I-IOT02A to your host computer using the USB port. Put
+the SW4 (MCU SWD) in OFF mode and SW5 (SWD BLE) in ON mode. Then build
+and flash an application. Here is an example for the
+:ref:`hci_uart <bluetooth-hci-uart-sample>` application.
+
+Run a serial host program to connect with your B-U585I-IOT02A board:
+
+.. code-block:: console
+
+   $ minicom -D /dev/ttyACM0
+
+Then build and flash the application for the STM32WB5MMG module.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/bluetooth/hci_uart
+   :board: stm32wb5mmg
+   :goals: build flash
+
+Next, reverse back the buttons to default mode (SW4 on ON and SW5
+on OFF) mode. In this case we will upload the Bluetooth sample on the
+main microcontroller.Then, build the bluetooth
+:zephyr_file:`samples/bluetooth/observer` demo application for
+B-U585I-IOT02A board:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/bluetooth/observer
```

Comment:
I had a success with this sample but with peripheral_hr, I got the following issue:
```
Bluetooth init failed (err -22)
```
Did you observe this as well ?

Reply:
That's because the connenction between BLE module and main micro controller doesn't have HW Flow control wiring. So, as it has been mentioned in the commit log, users should disable it with by adding `CONFIG_BT_HCI_ACL_FLOW_CONTROL=n` to prj.conf files. 

> commit 05d12f47f4c5757276d53d2518dad516ca4f93d1 (HEAD -> add_ble_support_b_u585i, origin/add_ble_support_b_u585i)
> Author: Javad Rahimipetroudi <javad.rahimipetroudi@mind.be>
> Date:   Mon Jan 15 15:01:40 2024 +0100
> 
>     boards: arm: b_u585i_iot02a: add bluetooth support
>     
>     This commits makes it possible to use the onboard bluetooth
>     module (STM32WB5MMG) with existing zephyr bluetooth samples.
>     Note that there was no hardware flow control wiring
>     available on the board, which is why it has been disabled
>     in the both main board and BLE module Device Tree. As the
>     board doesn't support HW flow control, users must set
>     CONFIG_BT_HCI_ACL_FLOW_CONTROL=n in project files.
console output for `peripheral_hr` after disabling `CONFIG_BT_HCI_ACL_FLOW_CONTROL` :
```
*** Booting Zephyr OS build zephyr-v3.5.0-4217-g05d12f47f4c5 ***
[00:00:00.051,000] <inf> bt_hci_core: Identity: 02:80:E1:00:00:00 (public)
[00:00:00.052,000] <inf> bt_hci_core: HCI: version 1.0b (0x00) revision 0xa077, manufacturer 0x0030
[00:00:00.052,000] <inf> bt_hci_core: LMP: version 1.0b (0x00) subver 0x2177
Bluetooth initialized
Advertising successfully started

```